### PR TITLE
Add new indexes for reindex

### PR DESCRIPTION
**Pull request by AmitPhulera**
This PR adds migrations to create new indexes for the upcoming reindex.

The migrations were generated by running

```
./manage.py make_elastic_migration --name make_new_indexes -c apps -c cases -c case_search -c domains -c forms -c groups -c sms -c users
```

Review by 🐟 

## Safety Assurance
Will create new indexes, which would not affect anything yet on HQ.
### Safety story

Pretty safe.
### Automated test coverage

NA
### QA Plan

NA

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

The migrations should be reverted if the PR is reverted.

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/tests/nose.py [hunk 1]
```diff
@@ -297,11 +297,12 @@ def teardown(self):
 
         self.blob_db.close()
 
+        self.delete_elastic_indexes()
```
Comment (millerdev) [inline corehq/tests/nose.py:300]: How much time does this typically take? If it takes significant time (or if it results in significant database setup time in the next test run) we may want to revert 179195d3d700587390aad5270c956304599d1b8a and find a different solution.

Another reason we may want to find a different solution would be if this causes unintuitive behavior for developers because `reusedb` has inconsistent behavior for different storage backends.

File: corehq/apps/es/utils.py [hunk 2]
```diff
@@ -186,4 +186,6 @@ def mapping_sort_key(item):
 
 def index_runtime_name(name):
     # transform the name if testing
-    return f"{TEST_DATABASE_PREFIX}{name}" if settings.UNIT_TESTING else name
+    if settings.UNIT_TESTING and not name.startswith(TEST_DATABASE_PREFIX):
```
Comment (millerdev) [inline corehq/apps/es/utils.py]: Given the eventual solution to getting tests passing on this PR, is this change still necessary?

I think it would be good to revert 4ccb3ee8281b2c3e2b5e7549e7ff92acf386400e if possible to make the outcome of this function more predictable in testing scenarios.

File: corehq/tests/nose.py (hunk 1 above)
Comment (AmitPhulera) [reply corehq/tests/nose.py:300]: I would prefer not reverting it because
1. deleting ES indexes is pretty fast, for this case on my local system deleting of every index just took .2 seconds.
2. Setup and deleting of every ES index happens automatically by @es_test decorator. 

So, I think this optimisation would not benefit us in similar terms as it does in couch and sql.

File: corehq/apps/es/utils.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/es/utils.py]: I explored the idea of reverting https://github.com/dimagi/commcare-hq/commit/4ccb3ee8281b2c3e2b5e7549e7ff92acf386400e and spend some good time into it but it was not straightforward to do.

`CreateIndex` index and `DeleteIndex` are related and they call each other in their rollbacks so the index name was getting appended by `test_`. And if we remove the prefixing in `CreateIndex` and `DeleteIndex` then it is failing where they are directly getting called.

Comment (millerdev) [reply corehq/apps/es/utils.py]: How about a solution like https://github.com/dimagi/commcare-hq/pull/33087?

I'd like to avoid conditionally adding the prefix because it requires callers to be aware of that conditionality and make sure they do not pass names that could cause unpredictable outcomes. For example, imagine a production index that started with `test_`.

Comment (AmitPhulera) [reply corehq/apps/es/utils.py]: > How about a solution like https://github.com/dimagi/commcare-hq/pull/33087?

Thanks for the suggestion. Looks like it could work. I can take a stab at the approach to see how it works.

> I'd like to avoid conditionally adding the prefix because it requires callers to be aware of that conditionality and make sure they do not pass names that could cause unpredictable outcomes. For example, imagine a production index that started with test_.

I understand your concern here. I tried to make it work as it is yesterday and failed. I will try with approach you suggested.

**Comment by AmitPhulera** [comment]
@millerdev I have created https://github.com/dimagi/commcare-hq/pull/33102 which implements the solution that we discussed yesterday. 
I was not able to update `make_elastic_migration` to add the `index_runtime_name` dynamically as it required understanding django migration framework. Given time constraint I went ahead with manually adding it to the migrations.
I am closing out this PR in favour of the one mentioned above.
